nodecontrol/kmdControl.go

```diff
@@ -200,7 +200,7 @@ func (kc *KMDController) StartKMD(args KMDStartArgs) (alreadyRunning bool, err e
 			logging.Base().Errorf("%s: kmd data dir exists but is not a directory", kc.kmdDataDir)
 			return false, errors.New("bad kmd data dir")
 		}
-		if (dataDirStat.Mode() & 0077) != 0 {
+		if !isDirectorySafe(dataDirStat) {
```

Comment:
add the logging line to the non-windows implementation. ( so that all the special permissions testing and logging would remain there )

Reply:
Hi @tsachiherman , I don't understand what do you mean with this. The `kmd data dir exists but is too permissive` is still there.

Reply:
what I meant was to move the line
```
logging.Base().Errorf("%s: kmd data dir exists but is too permissive (%o), change to (%o)", kc.kmdDataDir, dataDirStat.Mode()&0777, DefaultKMDDataDirPerms)
```
from kmdControl.go into kmdControl_common.go, so that all directory-mode bit masks would be handled in there.

Reply:
Got it. Tomorrow morning I'll do all the changes